feat: update twig to 3.18.0 (#6205)
shopware/platform: UPGRADE-6.8.md

# 6.8.0.0

# Changed Functionality

# API

# Core

# Administration

# Storefront

# App System

> ## Use `sw_macro_function` instead of usual `macro` in app scripts if you return values

Reviewer: I am not sure, if this is the right place :thinking: 

using this will already be possible with the next 6.6 and people should use this. so this should also be part of the upgrade guide of 6.6. 

it can be added to your changelog file under `# Upgrade Information`

Reviewer 2: Intended that way for now, it will only break in 6.8, but yeah heads up that it is now possible would be good 👍

Return values over the `return` keyword from usual twig `macro` functions are not supported anymore.

> Use the `sw_macro_function` instead, the new tag was available since v6.6.10.0.

Reviewer: ```suggestion
Use the `sw_macro_function` instead, which is available since v6.6.10.0.
```

```diff
// Resources/scripts/include/media-repository.twig
- {% macro getById(mediaId) %}

> + {% sw_macro_function getById(mediaId) %}

Reviewer: `sw_macro_function` is our functionality still a macro? or is there a better name, which represents more what our intention with this is?

Author: In the end, it is a macro. So, for me, the naming is fine.

Reviewer: yeah, the underlying used functionality in Twig is the macro, but would you name the thing like this, if you would invent it new? :grin: just nitpicking, no blocking. I was just curious
